src/overlays/actors/ovl_Bg_Hidan_Curtain/z_bg_hidan_curtain.c

```diff
@@ -21,25 +45,201 @@ const ActorInit Bg_Hidan_Curtain_InitVars = {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgHidanCurtain* this = THIS;
+    BgHidanCurtainParams* hcParams = &sHCParams[this->sizeIndex];
+    f32 riseStatus;
+
+    if ((globalCtx2->cameraPtrs[0]->setting == 0x28) || (globalCtx2->cameraPtrs[0]->setting == 0x38)) {
+        this->collider.base.atFlags &= ~2;
+    } else {
+        if (this->collider.base.atFlags & 2) {
+            this->collider.base.atFlags &= ~2;
+            func_8002F71C(globalCtx2, &this->actor, 5.0f, this->actor.yawTowardsLink, 1.0f);
+        }
+        if ((this->type == 4) || (this->type == 5)) {
+            this->actor.posRot.pos.y =
+                (2.0f * this->actor.initPosRot.pos.y) - hcParams->riseDist - this->actor.posRot.pos.y;
+        }
+
+        this->actionFunc(this, globalCtx2);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Curtain/func_80885434.s")
+        if ((this->type == 4) || (this->type == 5)) {
+            this->actor.posRot.pos.y =
+                (2.0f * this->actor.initPosRot.pos.y) - hcParams->riseDist - this->actor.posRot.pos.y;
+        }
+        riseStatus =
+            (hcParams->riseDist - (this->actor.initPosRot.pos.y - this->actor.posRot.pos.y)) / hcParams->riseDist;
+        this->alpha = 255.0f * riseStatus;
+        if (this->alpha > 50) {
+            this->collider.dim.height = hcParams->height * riseStatus;
+            CollisionCheck_SetAT(globalCtx2, &globalCtx2->colChkCtx, &this->collider.base);
+            CollisionCheck_SetOC(globalCtx2, &globalCtx2->colChkCtx, &this->collider.base);
+            if (gSaveContext.sceneSetupIndex <= 3) {
+                func_8002F974(&this->actor, NA_SE_EV_FIRE_PILLAR_S - SFX_FLAG);
+            }
+        } else if ((this->type == 1) && Flags_GetTreasure(globalCtx2, this->treasureFlag)) {
+            Actor_Kill(&this->actor);
+        }
+        this->texScroll++;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Curtain/func_808854E8.s")
+void BgHidanCurtain_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanCurtain* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Curtain/func_80885514.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_hidan_curtain.c", 685);
+    func_80093D84(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Curtain/func_80885554.s")
+    gDPSetPrimColor(oGfxCtx->polyXlu.p++, 0x80, 0x80, 255, 220, 0, this->alpha);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Curtain/func_80885594.s")
+    gDPSetEnvColor(oGfxCtx->polyXlu.p++, 255, 0, 0, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Curtain/func_80885604.s")
+    gSPSegment(oGfxCtx->polyXlu.p++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, this->texScroll & 0x7F, 0, 0x20, 0x40, 1, 0,
+                                (u8)(this->texScroll * -0xF), 0x20, 0x40));
```

Comment:
This is slightly better imo (if it matches)
```suggestion
                                (this->texScroll * -0xF) & 0xFF, 0x20, 0x40));
```